internal/fs/handle/dir_handle.go

```diff
@@ -182,11 +184,26 @@ func readAllEntries(
 		}
 	}
 
+	// Append local file entries (not synced to GCS).
+	localEntries := in.LocalFileEntries(localFileInodes)
+	entries = append(entries, localEntries...)
+
 	// Ensure that the entries are sorted, for use in fixConflictingNames
 	// below.
 	sort.Sort(sortedDirents(entries))
 
 	// Fix name conflicts.
+	// When a local file is synced to GCS but not removed from the local file map,
+	// the entries list will have two duplicate entries.
+	// To handle this scenario, we had 2 options:
+	// Option 1: [Selected]
+	// Throw an error while fixing conflicting names. The error will be fixed in
+	// subsequent ls calls.
```

Comment:
this error will be fixed in subsequent ls calls assuming that entry will be removed from localFileInodes.